localization/pose_estimator/ndt_scan_matcher/ndt_omp/CMakeLists.txt

```diff
@@ -1,70 +1,87 @@
-cmake_minimum_required(VERSION 3.0.2)
+cmake_minimum_required(VERSION 3.5)
 project(ndt_omp)
 
-# -mavx causes a lot of errors!!
-add_definitions(-std=c++14 -msse -msse2 -msse3 -msse4 -msse4.1 -msse4.2)
-set(CMAKE_CXX_FLAGS "-std=c++14 -msse -msse2 -msse3 -msse4 -msse4.1 -msse4.2")
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
 
 # warn performance issue of ndt_omp which is NOT built in release mode
-string(TOUPPER ${CMAKE_BUILD_TYPE} uppercase_CMAKE_BUILD_TYPE)
-if (NOT (${uppercase_CMAKE_BUILD_TYPE} STREQUAL "RELEASE"))
-  message(WARNING
-    "CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} may cause a serious performance degradation. Please configure with -DCMAKE_BUILD_TYPE=Release.")
+if (NOT ("${CMAKE_BUILD_TYPE}" STREQUAL "Release"))
+  message(WARNING "CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} may cause a serious performance degradation. Please configure with -DCMAKE_BUILD_TYPE=Release.")
 endif()
 
-find_package(catkin REQUIRED COMPONENTS
-  roscpp
-  pcl_ros
-)
-
+find_package(ament_cmake REQUIRED)
+find_package(PCL REQUIRED COMPONENTS common filters kdtree registration io)
 find_package(OpenMP)
+
 if (OPENMP_FOUND)
   set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
   set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
 endif()
 
-catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES ndt_omp
-  CATKIN_DEPENDS
-)
-
 ###########
 ## Build ##
 ###########
 
-include_directories(
-  include
-  ${catkin_INCLUDE_DIRS}
-)
-
 add_library(ndt_omp
   src/ndt_omp/voxel_grid_covariance_omp.cpp
   src/ndt_omp/ndt_omp.cpp
 )
 
-add_executable(align
-  apps/align.cpp
-)
-add_dependencies(align
-  ndt_omp
+target_include_directories(ndt_omp
+  PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include> 
 )
+
 # pcl 1.7 causes a segfault without -O2 (or -O3) flag
```

Comment:
Just a comment - the default installation of pcl for Ubuntu 20.04 is 1.10. Are the compile options still necessary? I guess you do more here with the msse options though so easier to leave as is?

Also for me installing pcl required the package libpcl-dev, but I see below you install libpcl-all-dev, I think I remember reading about this issue, but if you need libpcl-all-dev and had to do something to find it, can you please add in these comments for a record.

Reply:
That's a good question that's difficult to answer. The comment doesn't say which part segfaults, or whether PCL segfaults immediately, so it's hard to reproduce, but I'd expect such a serious bug to be fixed by now. There seems to be no open issue about this either on the PCL project's GitHub. I'm fine with removing it and if PCL segfaults, we can check back.

Reply:
I think `libpcl-all-dev` is the "ROS name" (not sure what the name for that abstraction layer is) for the `libpcl-dev` package on Ubuntu: https://github.com/ros/rosdistro/blob/f8335c214e20d445756ba6479e69a050bbeb3094/rosdep/base.yaml#L3429

Reply:
I tried out the `align` binary and it doesn't segfault even when the build type is not set to Release, so it should be fine.